test/linalg/lu.jl

```diff
@@ -0,0 +1 @@
+@test logdet(Complex64[1.0f0 0.5f0; 0.5f0 -1.0f0]) === 0.22314355f0 + 3.1415927f0im
```

Comment:
The other LU tests are at different places in linalg1-4.jl and at some point they should go into the new lu.jl, but I don't think we should hold back this fix by requiring the reorganization to take place there. We should probably also add some more LU test.